File: 3447-clear-digits/3447-clear-digits.c
char* clearDigits (char* s) {
    int len = strlen (s), top = -1;
    char* stack = (char*) malloc (len + 1);
    for (int i = 0; i < len; i++) {
        if (isdigit (s [i])) {
            if (top >= 0) top--;
        }
        else
            stack [++top] = s [i];
    }
    stack [++top] = '\0';
    return stack;
}

File: 3447-clear-digits/3447-clear-digits.java
class Solution {
    public String clearDigits (String s) {
        StringBuilder stack = new StringBuilder ();
        for (char c : s.toCharArray ()) {
            if (Character.isDigit (c)) {
                if (stack.length () > 0) {
                    stack.deleteCharAt (stack.length () - 1);
                }
            } else {
                stack.append (c);
            }
        }
        return stack.toString ();
    }
}